### PR TITLE
Add test proxy init fixture

### PR DESCRIPTION
**Pull request by aaron-lane**
## Background

This branch adds the test proxy init fixture with initial support for Ubuntu on GCP deployments.

Asana task: https://app.asana.com/0/1181500399442529/1201422598987197/f

## How has this been tested?

Tested here https://github.com/hashicorp/terraform-google-terraform-enterprise/pull/135

## This PR makes me feel

optional gif describing your feelings about this pr

### CODE REVIEW

File: fixtures/test_proxy_init/templates/mitmproxy.sh.tpl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+curl --location --remote-name https://snapshots.mitmproxy.org/6.0.2/mitmproxy-6.0.2-linux.tar.gz
+tar --extract --file ./mitmproxy*.tar.gz -C /usr/local/bin/
+confdir="/etc/mitmproxy"
+mkdir --parents $confdir
+
+%{ if ca_certificate_secret != "" && ca_private_key_secret != "" ~}
+echo "[$(date +"%FT%T")] Deploying certificates for mitmproxy" | tee --append /var/log/ptfe.log
+certificate="$confdir/mitmproxy-ca.pem"
+
+gcloud secrets versions access latest --secret="${ca_certificate_secret}" \
```
Comment (anniehedgpeth) [inline fixtures/test_proxy_init/templates/mitmproxy.sh.tpl:18]: Do we want to add a variable for cloud provider here with an if statement, assuming the others would be added as the support for the other clouds is added? i.e. `%{ if cloud_provider == "gcp" ~}`

File: fixtures/test_proxy_init/main.tf [hunk 2]
```diff
@@ -1 +1,17 @@
-# coming soon
\ No newline at end of file
+locals {
+  mitmproxy_http_port = 8080
+  mitmproxy_user_data_script = templatefile(
```
Comment (anniehedgpeth) [inline fixtures/test_proxy_init/main.tf:3]: This reminds me that I need to holler at security to let us use the GHA for `terraform fmt`. :( Would you mind running that locally?

File: fixtures/test_proxy_init/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+resource "google_compute_instance" "squid" {
+  boot_disk {
+    initialize_params {
+      image = data.google_compute_image.ubuntu.id
+    }
+  }
+
+  name         = "squid"
+
+  metadata_startup_script = module.test_proxy_init.squid.user_data_script
```
Comment (anniehedgpeth) [inline fixtures/test_proxy_init/README.md:35]: Nice and simple. I like it.

**Review by anniehedgpeth — APPROVED** [review]
Thank you for adding this! Excited to see this repo getting some love. Just a couple of non-blocking things...

File: fixtures/test_proxy_init/templates/mitmproxy.sh.tpl (hunk 1 above)
Comment (aaron-lane) [reply fixtures/test_proxy_init/templates/mitmproxy.sh.tpl:18]: Yes, though I opted to leave the implementation for the contributor that adds the additional platform support. I didn't want to predetermine what that might need to look like.

File: fixtures/test_proxy_init/main.tf (hunk 2 above)
Comment (aaron-lane) [reply fixtures/test_proxy_init/main.tf:3]: Turns out, this is already formatted according to `terraform fmt`. I think it chooses to not align arguments when one has an associated "block" body.

**Pull request merged**